### PR TITLE
User synchronizer

### PR DESCRIPTION
**Pull request by Gerhut**
TODO:

- [x] deploy

### CODE REVIEW

File: src/ClusterBootstrap/services/user-synchronizer/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+              value: '{{cnf["user-synchronizer"]["client_secret"]}}'
+            - name: WINBIND_URL
+              value: '{{cnf["user-synchronizer"]["winbind_url"]}}'
+            - name: DATABASE_URL
+              value: 'mysql://{{cnf["mysql_username"]}}:{{cnf["mysql_password"]}}@{{cnf["mysql_node"]}}:{{cnf["mysql_port"]}}/DLWSCluster-{{cnf["clusterId"]}}'
+            - name: GROUPS_ID
+              value: '{{cnf["user-synchronizer"]["groups_id"]}}'
+          nodeSelector:
+            user-synchronizer: active
+          restartPolicy: Never
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/services/user-synchronizer/cronjob.yaml]: Do we not want to restart upon failure?

File: src/user-synchronizer/logger.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from logging import getLogger, FileHandler, StreamHandler
```
Comment (Anbang-Hu) [inline src/user-synchronizer/logger.py]: Let's use configuration file: https://github.com/microsoft/DLWorkspace/blob/dltsdev/src/RestAPI/logging.yaml

File: src/user-synchronizer/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+## Get Started
+
+```shell
+pipenv install
+pipenv run start
+```
+
+## Configuration
+
+It uses environment variables to configure the behavior of the service:
```
Comment (Anbang-Hu) [inline src/user-synchronizer/README.md]: Not sure which is a better option - using environment variables or using a yaml configuration file.

File: src/user-synchronizer/__main__.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        'gid': int(identity['gid']),
+        'groups': dumps(identity['groups'], separators=(',', ':')),
+    }
+    response = get(restfulapi_url + '/AddUser', params=params)
+    response.raise_for_status()
+
+    logger.info('Updated {} to cluster'.format(user_name))
+
+
+for group_identity in iter_group_identities():
```
Comment (Anbang-Hu) [inline src/user-synchronizer/__main__.py]: Periodic run?

File: src/user-synchronizer/README.md (hunk 3 above)
Comment (Gerhut) [reply src/user-synchronizer/README.md]: Either is OK, I am just following [the 12 factors](https://12factor.net/config)

**Pull request merged**